src/soroban/parsers.ts

```diff
@@ -7,10 +7,18 @@ export function parseRawSendTransaction(
   const errResult = r.errorResultXdr;
   delete r.errorResultXdr;
 
+  const diagEvents = r.diagnosticEventsXdr;
+  delete r.diagnosticEventsXdr;
+
   if (!!errResult) {
     return {
       ...r,
-      errorResult: xdr.TransactionResult.fromXDR(errResult, 'base64')
+      ...(diagEvents !== undefined && diagEvents.length > 0 && {
+        diagnosticEvents: diagEvents.map(
```

Comment:
I haven't loaded this locally in ide for ts, but does this require adding `diagnosticEvents` to the `SendTransactionResponse` interface in api.ts?